[ADD] PLM: engineering change order doc (#5238)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/plm/engineering_change_orders.rst

=========================
Engineering change orders
=========================

.. _plm/eco:

Utilize *engineering change orders* (ECO) to track and implement changes to products and :ref:`bill

> of materials <manufacturing/management/bill-configuration>`. View and create :abbr:`ECOs

Author: Oh yes! thanks for catching that

clicking `ECO00X` in the breadcrumbs in the top left. On the :abbr:`ECO (engineering change order)`
form, a new :guilabel:`Operation Changes` tab displays the differences between the current
production :abbr:`BoM (Bill of Material)` and the new version. Modifications to the :abbr:`BoM (Bill
of Material)` in an :abbr:`ECO (engineering change order)` will not affect the :abbr:`BoM (Bill of
Material)` used in production. That is, until the :ref:`changes are applied
<plm/eco/apply-changes>`.

.. _plm/eco/apply-changes:

Apply changes
=============

After verifying the changes, move the :abbr:`ECO (engineering change order)` to a :ref:`verification
stage <plm/eco/stage-config>`. Once the approvers accept the changes, the :guilabel:`Apply Changes`
button will become available. Click this button, and the :abbr:`ECO (engineering change order)`
will be automatically closed.

To ensure the changes are live, navigate to the :guilabel:`Revision` smart button. The :abbr:`BoM
(Bill of Material)` will have the large red :guilabel:`Archived` tag removed. Additionally, navigate
to the product form, switch to the :guilabel:`Miscellaneous` tab, and the :guilabel:`Version` field

> will be incremented.

Reviewer: It might be good to give a more detailed explanation than just saying "incremented".

   the following section.

   .. image:: engineering_change_orders/add-a-step.png
      :align: center
      :alt: Fill out the *Add a Step* form to suggest an additional quality control point.

View ECO
--------

To review the proposed changes, go to the `BOM Changes` ECO type from :menuselection:`PLM app -->
Overview` and click on the card to view the contained :abbr:`ECOs (engineering change orders)`.
Click on the newly created :abbr:`ECO (engineering change order)` in the `New` stage, and view a
summary of the proposed changes in the :guilabel:`Operation Changes` tab. To implement the changes,
click the :guilabel:`Revision` smart button.

.. example::
   The :abbr:`ECO (engineering change order)` adding another check for broken components is created
   in the `BOM Changes` ECO type found in :menuselection:`PLM app --> Overview`. By default,
   :abbr:`ECOs (engineering change orders)` created from tablet view are named with the
   manufacturing order number  as a suggested form of reference.

   .. image:: engineering_change_orders/view-BOM-change.png
      :align: center
      :alt: alt text

> In the new :abbr:`BoM (Bill of Material)`, switch to the :guilabel:`Operations` tab and select the

Reviewer: ```suggestion
On the new :abbr:`BoM (Bill of Material)`, switch to the :guilabel:`Operations` tab and select the
```

> :guilabel:`Show Instructions` icon (:guilabel:`☰ (three horizontal lines)`). Doing so opens a list

Reviewer: ```suggestion
:guilabel:`☰ (Show Instructions)` icon. Doing so opens a list
```
I like to use the symbol itself followed by the name of the symbol in parentheses (ex. settings).

of :guilabel:`Steps` to perform the operation, with the newest instruction titled `New Step
Suggestion:` followed by the user-entered title. Click the line item to view the suggested changes.

.. image:: engineering_change_orders/show-instructions.png
   :align: center
   :alt: "Show Instructions" icon in the *Operations* tab of a BoM.

> In opened the :ref:`quality control point <quality/quality_control_points>` form, ensure the

Reviewer: ```suggestion
On the :ref:`quality control point <quality/quality_control_points>` form, ensure the
```

=========================
Engineering change orders
=========================

.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
.. |BOMs| replace:: :abbr:`BoMs (Bills of Materials)`
.. |ECO| replace:: :abbr:`ECO (Engineering Change Order)`
.. |ECOs| replace:: :abbr:`ECOs (Engineering Change Orders)`

.. _plm/eco:

Utilize *Engineering Change Orders* (ECO) to track and implement changes to products and :ref:`bills
of materials <manufacturing/management/bill-configuration>`. View and create :abbr:`ECOs
(engineering change orders)` within each *ECO type*, a project pipeline in Gantt view that consists
of stages, ensuring progress tracking for collaborators and stakeholders.

Engineering change orders can be created:

> #. :ref:`directly in the ECO type <plm/eco/create-eco>`
> #. by an operator in the :ref:`tablet view <plm/eco/tablet-view>` of an operation
> #. automatically from feedback submitted to the :ref:`ECO type's email alias <plm/eco/eco-type>`.

Author: ```suggestion
#. :ref:`directly in the ECO type <plm/eco/create-eco>`.
#. by an operator in the :ref:`tablet view <plm/eco/tablet-view>` of an operation.
#. automatically from feedback submitted to the :ref:`ECO type's email alias <plm/eco/eco-type>`.
```

operations by clicking the :guilabel:`Add a line` button, and remove new operations by clicking the
:guilabel:`Archive` button.

.. note::
   The :guilabel:`Operations` tab is not available by default. To enable it, navigate to
   :menuselection:`Manufacturing --> Configuration --> Settings` and check the :guilabel:`Work
   Orders` box.

Once the changes are complete, navigate back to the |ECO| itself by clicking `ECO00X` in the
breadcrumbs in the top left. On the |ECO| form, a new :guilabel:`Operation Changes` tab displays the
differences between the current production |BOM| and the new version. Modifications to the |BOM| in
an |ECO| will not affect the |BOM| used in production. That is, until the :ref:`changes are applied
<plm/eco/apply-changes>`.

.. _plm/eco/apply-changes:

Apply changes
=============

After verifying the changes, move the |ECO| to a :ref:`verification stage <plm/eco/stage-config>`.
Once the approvers accept the changes, the :guilabel:`Apply Changes` button will become available.
Click this button, and the |ECO| will be automatically closed and the changes are applied.

> To ensure the changes are live, navigate to the :guilabel:`Revision` smart button. The |BOM| will
> have the large red :guilabel:`Archived` tag removed. Additionally, navigate to the product form,
> switch to the :guilabel:`Miscellaneous` tab, and the :guilabel:`Version` field will be updated to
> match the version number shown on the :guilabel:`Revision` smart button of the latest |ECO|.

Reviewer 2: ```suggestion
To ensure the changes are live, navigate to the :guilabel:`Revision` smart button.

The |BOM| will have the large red :guilabel:`Archived` tag removed. Additionally, navigate to the
product form, switch to the :guilabel:`Miscellaneous` tab, and the :guilabel:`Version` field will be
updated to match the version number shown on the :guilabel:`Revision` smart button of the latest
|ECO|.
```

.. example::
   After applying the changes of the |ECO| for the :ref:`keyboard <plm/eco/example-keyboard>`, view
   the version of the current keyboard |BOM| in the :guilabel:`Miscellaneous` tab. Here, the
   :guilabel:`Version` number has been updated to `2`, matching the `V2` that appears in the
   :guilabel:`Revision` smart button of the |ECO|.

   .. image:: engineering_change_orders/BOM-version.png
      :align: center
      :alt: View current *BOM* version in the Miscellaneous tab.

.. _plm/eco/tablet-view:

Create ECO from tablet view
===========================

|ECOs| created in *tablet view* are intended for operators to directly suggest clearer operation
instructions. To create an |ECO|, navigate to the :guilabel:`Work Orders` tab on an ongoing

> manufacturing order. Click the :guilabel:`tablet icon` for the desired work order to open the tablet
> view of the operation.

Reviewer 2: ```suggestion
manufacturing order. Then, click the :guilabel:`tablet icon` for the desired work order to open the
tablet view of the operation.
```

.. image:: engineering_change_orders/tablet-icon.png
   :align: center
   :alt: Find the tablet icon for each operation, second from the far right.

> Next, add an instructional step by clicking the :guilabel:`☰ (three horizontal lines)` icon in
> tablet view of an operation. Then, click the :guilabel:`Add a step` button.

Reviewer 2: ```suggestion
Next, add an instructional step, by clicking the :guilabel:`☰ (three horizontal lines)` icon in the
tablet view of an operation. Then, click the :guilabel:`Add a step` button.
```

.. image:: engineering_change_orders/additional-options-menu.png
   :align: center
   :alt: Navigate to the "Add a Step" pop-up by clicking the three horizontal lines icon in tablet
         view.

In the :guilabel:`Title` field, enter a short step description. Next, in the
:guilabel:`Instructions` text field, type the instructions of the step in greater detail.
Optionally, add an image to the :guilabel:`Document` field. Once completed, finish by clicking the
:guilabel:`Propose Change` button.

.. example::
   To propose an additional check for broken components, enter the details in the :guilabel:`Add a

>    Step` pop-up. Doing so creates an instructional quality control point that will be reviewed in
>    the following section.

Reviewer 2: ```suggestion
   Step` pop-up window. Doing so creates an instructional quality control point that will be
   reviewed in the following section.
```

   .. image:: engineering_change_orders/add-a-step.png
      :align: center
      :alt: Fill out the *Add a Step* form to suggest an additional quality control point.

View ECO
--------

To review the proposed changes, go to the `BOM Changes` ECO type from :menuselection:`PLM app -->

> Overview` and click on the card to view the contained |ECOs|. Click on the newly created |ECO| in
> the `New` stage, and view a summary of the proposed changes in the :guilabel:`Operation Changes`
> tab. To implement the changes, click the :guilabel:`Revision` smart button.

Reviewer 2: ```suggestion
Overview`, and click on the card to view the contained |ECOs|. Click on the newly created |ECO| in
the `New` stage, and view a summary of the proposed changes in the :guilabel:`Operation Changes`
tab. To implement the changes, click the :guilabel:`Revision` smart button.
```

.. example::

   An |ECO| adding another check for broken components is created in the `BOM Changes` ECO type
   found in :menuselection:`PLM app --> Overview`. By default, |ECOs| created from tablet view are
   named with the manufacturing order number for reference.

   .. image:: engineering_change_orders/view-BOM-change.png
      :align: center
      :alt: Find the new ECO in the "BOM Changes" ECO type, in the *New* stage.

> On the new |BOM|, switch to the :guilabel:`Operations` tab and select the :guilabel:`☰ (Show
> Instructions)` icon. Doing so opens a list of :guilabel:`Steps` to perform the operation, with the
> newest instruction titled `New Step Suggestion:` followed by the user-entered title. Click the line
> item to view the suggested changes.

Reviewer 2: ```suggestion
On the new |BOM|, switch to the :guilabel:`Operations` tab, and select the :guilabel:`☰ (Show
Instructions)` icon. Doing so opens a list of :guilabel:`Steps` to perform the operation, with the
newest instruction titled `New Step Suggestion:`, followed by the user-entered title. Click the line
item to view the suggested changes.
```

.. image:: engineering_change_orders/show-instructions.png
   :align: center
   :alt: "Show Instructions" icon in the *Operations* tab of a BoM.

On the :ref:`quality control point <quality/quality_control_points>` form, ensure the following form
fields are accurately filled out to give detailed instructions for operators:

> - :guilabel:`Title`: rename to give a concise description of the new instruction
> - :guilabel:`Control per`: using the drop-down menu, determine whether this instruction applies
>   broadly for the :guilabel:`Product`, specifically for this :guilabel:`Operation` only, or a
>   particular :guilabel:`Quantity` of the product
> - :guilabel:`Type`: categorizes the control point type. From the drop-down menu, select
>   :guilabel:`Instructions` to detail an instruction for the worker. To receive input from the
>   workers, select the :guilabel:`Take a Picture`, :guilabel:`Register Consumed Materials`,
>   :guilabel:`Print Label`, or other :ref:`quality check options <quality/quality_control_points>`.

Reviewer 2: ```suggestion
- :guilabel:`Title`: rename to give a concise description of the new instruction.
- :guilabel:`Control per`: using the drop-down menu, determine whether this instruction applies
  broadly for the :guilabel:`Product`, specifically for this :guilabel:`Operation` *only*, or a
  particular :guilabel:`Quantity` of the product.
- :guilabel:`Type`: categorizes the control point type. From the drop-down menu, select
  :guilabel:`Instructions` to detail an instruction for the worker. To receive input from the
  workers, select the :guilabel:`Take a Picture`, :guilabel:`Register Consumed Materials`,
  :guilabel:`Print Label`, or other :ref:`quality check options <quality/quality_control_points>`.
```

Once the quality control point is configured, return to the :guilabel:`Steps` list using the
breadcrumbs. Finally, drag the last quality control line item to its intended order of instructions.

.. example::

>    Drag and reorder the `Check for broken switches` instruction by clicking and dragging its "6
>    dots" icon to move it from the bottom to the second position.

Reviewer 2: ```suggestion
   Drag and reorder the `Check for broken switches` instruction, by clicking and dragging its "6
   dots" icon to move it from the bottom to the second position.
```

   .. image:: engineering_change_orders/reorder.png
      :align: center
      :alt: Drag and reorder instructions by selecting the "6 dots" icon on the far left.

.. _plm/eco/eco-type:

Create ECO type
===============

> To access and manage ECO types, navigate to :menuselection:`PLM app --> Configuration --> ECO
> Types`. Create a new ECO type by clicking :guilabel:`New`, and fill in the :guilabel:`Name`. The
> :guilabel:`Email Alias` field is optional, but feedback submitted to the filled-in email address
> will automatically create an |ECO| in this ECO type. Modify existing ECO type names and email
> aliases by clicking on the respective ECO type in the list.

Reviewer 2: ```suggestion
To access and manage ECO types, navigate to :menuselection:`PLM app --> Configuration --> ECO
Types`.

Create a new ECO type by clicking :guilabel:`New`, and fill in the :guilabel:`Name`. The
:guilabel:`Email Alias` field is optional, but feedback submitted to the filled-in email address
will automatically create an |ECO| in this ECO type.

Modify existing ECO type names and email aliases by clicking on the respective ECO type in the list.
```

> Existing ECO types are displayed in task form on the :guilabel:`PLM Overview` page, accessible by
> navigating to :menuselection:`PLM app --> Overview`.

Reviewer 2: ```suggestion
Existing ECO types are displayed in the task form, on the :guilabel:`PLM Overview` page, which is
accessible by navigating to :menuselection:`PLM app --> Overview`.
```

.. _plm/eco/stage-config:

Stage configuration
-------------------

> Click an ECO type from :menuselection:`PLM app --> Overview` to open a kanban view of |ECOs| of this
> type. In the pipeline, ensure there is at least one *verification* stage, where |ECO| changes can be
> applied, and a *closing stage*, for storing completed |ECOs|. To configure a stage, hover over the
> intended stage and select the :guilabel:`⚙️ (gear)` icon. Then, click :guilabel:`Edit` to open a
> pop-up window.

Reviewer 2: ```suggestion
Click an ECO type from :menuselection:`PLM app --> Overview` to open a kanban view of |ECOs| of this
type.

In the pipeline, ensure there is at least one *verification* stage, where |ECO| changes can be
applied, and a *closing stage*, for storing completed |ECOs|.

To configure a stage, hover over the intended stage, and select the :guilabel:`⚙️ (gear)` icon.
Then, click :guilabel:`Edit` to open a pop-up window.
```

> Configure the verification stage in the edit stage pop-up by checking the box for :guilabel:`Allow
> to apply changes`. Optionally, add :guilabel:`Approvals` to automatically notify people to verify
> the changes before they go live. Once finished, select the :guilabel:`Save & Close` button.

Reviewer 2: ```suggestion
Configure the verification stage in the edit stage pop-up window, by checking the box for
:guilabel:`Allow to apply changes`.

Optionally, add :guilabel:`Approvals` to automatically notify people to verify the changes before
they go live. Once finished, select the :guilabel:`Save & Close` button.
```

.. example::

>    The stage named `Validated` is intended to store |ECOs| to be reviewed by the main approver, the
>    engineering manager, before the changes are applied to production. To reflect this, the
>    engineering manager is listed in the :guilabel:`Approvals` section. Additionally, the
>    :guilabel:`Allow to apply changes` option is checked.

Reviewer 2: ```suggestion
   The stage titled, `Validated,` is intended to store |ECOs| that will be reviewed by the main
   approver (the engineering manager) before the changes are applied to production. To reflect this,
   the engineering manager is listed in the :guilabel:`Approvals` section. Additionally, the
   :guilabel:`Allow to apply changes` option is checked.
```

   .. image:: engineering_change_orders/verification-stage.png
      :align: center
      :alt: Show "Allow to apply changes" option is checked.

> Next, hover over the closing stage and select the corresponding :guilabel:`⚙️ (gear)` icon to open
> the edit stage pop-up. Ensure that the :guilabel:`Folded in kanban view`, :guilabel:`Allow to apply
> changes`, and :guilabel:`Final Stage` options are checked.

Reviewer 2: ```suggestion
Next, hover over the closing stage, and select the corresponding :guilabel:`⚙️ (gear)` icon to open
the edit stage pop-up window. Ensure that the :guilabel:`Folded in kanban view`, :guilabel:`Allow to
apply changes`, and :guilabel:`Final Stage` options are checked.
```

=========================
Engineering change orders
=========================

.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
.. |BOMs| replace:: :abbr:`BoMs (Bills of Materials)`
.. |ECO| replace:: :abbr:`ECO (Engineering Change Order)`
.. |ECOs| replace:: :abbr:`ECOs (Engineering Change Orders)`

.. _plm/eco:

> Utilize *Engineering Change Orders* (*ECOs*) to track and implement changes to products, and

Reviewer 3: _Engineering change orders_ is not a proper noun.
Suggestion around specificity, as well (optional).

```suggestion
Utilize *engineering change orders* (*ECOs*) to track, implement, and revert change versions made to products, and
```

:ref:`bills of materials <manufacturing/management/bill-configuration>`.

> View and create :abbr:`ECOs (engineering change orders)` within each *ECO type*, a project pipeline

Reviewer 3: the short-hand notations at the top of the doc have the acronym definitions in title case, but here's it's not. Let's be consistent 😉

Also why not use the notation here?
```suggestion
View and create |ECOs| within each *ECO type*, a project pipeline
```

> in Gantt view with stages, ensuring progress tracking for collaborators and stakeholders.

Reviewer 3: Lines 15-16 have mixed subjects and don't offer a clear definition or an instruction.

Optionally, set a particular deadline for the completion of the change with an :guilabel:`Effective`
date, or add :guilabel:`Tags` for categorization.

Finally, click :guilabel:`Start Revision` after filling out the form to begin implementing the
changes. Clicking the button unveils the stages of the ECO type, the :guilabel:`Documents` smart
button, the :guilabel:`Revision` smart button, and the new |BOM| version number.

.. note::
   The :guilabel:`Revision` smart button is available only when the :guilabel:`Bill of Materials`
   radio button is selected in the :guilabel:`Apply on` field and the :guilabel:`Smart Revision` is
   activated.

.. image:: engineering_change_orders/eco-form.png
   :align: center
   :alt: Active ECO with overview of stages in the top right, and *Revision* smart button.

Change components
-----------------

To modify the components in a |BOM|, click the :guilabel:`Revision` smart button on an active |ECO|
to access the new version of the |BOM|. Odoo distinguishes the non-production version of the |BOM|
from the current version, by flagging the test |BOM| with a large :guilabel:`Archived` tag.

> On the new |BOM|, switch to the :guilabel:`Components` tab. Modify the components list by changing

Author: Hm, yes. Let me be more specific then!

Once the changes are complete, navigate back to the |ECO| itself, by clicking `ECO00X` in the
breadcrumbs in the top-left corner. On the |ECO| form, a new :guilabel:`BoM Changes` tab displays
the differences between the current |BOM| and the new version.

This ensures changes and tests are encapsulated in the revised |BOM| and do not affect the |BOM|
currently used in production. That is, until the :ref:`changes are applied <plm/eco/apply-changes>`.

.. example::
   View the summary of the differences between the current and revised keyboard |BOMs| in the
   :guilabel:`BoM Changes` tab of the |ECO|.

   .. image:: engineering_change_orders/BoM-changes.png
      :align: center
      :alt: View summary of component changes in the *BoM Changes* tab.

Change operations
=================

To modify the operations in a |BOM|, click the :guilabel:`Revision` smart button on an active |ECO|
to access the archived, new version of the |BOM|.

In the new |BOM| version, switch to the :guilabel:`Operation Changes` tab, and make changes to the
operations, by selecting each operation line item, and opening the corresponding :guilabel:`Open:

> Operations` pop-up window.

Author: Thank you @ksc-odoo for asking all of these questions! This helps me understand what additional contextual information I can provide to explain the process clearly. Voicing any and all uncertainty you have about the flow is super welcome! :grin: I'll address your questions in my next commit and point out the lines changed (:

As for the runbot version, I've been testing everything in runbot 16.4, though I'm uncertain whether this is the right move. To my understanding, 16.3 is 16.0 with spice, and 16.4 is the beta version of 17.0. So in actuality, this doc was written for the future version in mind. I'm not sure if this is best practice, as this document is technically for version 16. Thoughts, @StraubCreative ?

> Operations` pop-up window.

Author: Decided to remove the word "active". I initially used that adjective to describe ECOs where the "Start Revision" button has been pressed, initiating the BOM changed. But the ECOs don't spend a lot of time in the draft stage, so it didn't really make sense to make that distinction

> Operations` pop-up window.

Author: Also, KC! The "Operations" tab is correct. I didn't notice my error, so thank you for checking for me!

Make changes to any of the fields in the pop-up window, and click :guilabel:`Save` once completed.

Create new operations by clicking the :guilabel:`Add a line` button, and remove new operations by
clicking the :guilabel:`Archive` button.

.. note::
   The :guilabel:`Operations` tab is *not* available by default. To enable it, navigate to
   :menuselection:`Manufacturing app --> Configuration --> Settings`, and check the :guilabel:`Work
   Orders` box.

Once the changes are complete, navigate back to the |ECO| itself, by clicking `ECO00X` in the
breadcrumbs in the top-left corner.

On the |ECO| form, a new :guilabel:`Operation Changes` tab displays the differences between the
current production |BOM| and the new version. Modifications to the |BOM| in an |ECO| will *not*
affect the |BOM| used in production. That is, until the :ref:`changes are applied
<plm/eco/apply-changes>`.

.. _plm/eco/apply-changes:

Apply changes
=============

> After verifying the changes, move the |ECO| to a :ref:`verification stage <plm/eco/stage-config>`.

Author: Do you think this rewrite is sufficient? Otherwise, I'll think again about how to define this better.

> After verifying the changes, move the |ECO| to a :ref:`verification stage <plm/eco/stage-config>`,
which are stages that require approval before the revised changes can be applied to production
|BOM|.

Once the approvers accept the changes, the :guilabel:`Apply Changes` button becomes available. Click
this button, and the |ECO| is automatically closed, and the changes are applied.

To ensure the changes are live, navigate to the :guilabel:`Revision` smart button.

The |BOM| will have the large red :guilabel:`Archived` tag removed. Additionally, navigate to the

> product form, switch to the :guilabel:`Miscellaneous` tab, and the :guilabel:`Version` field will be

Author: Thank you for your thoughts! I'll improve my explanation for this part (:

updated to match the version number shown on the :guilabel:`Revision` smart button of the latest
|ECO|.

.. example::
   After applying the changes of the |ECO| for the :ref:`keyboard <plm/eco/example-keyboard>`, view
   the version of the current keyboard |BOM| in the :guilabel:`Miscellaneous` tab. Here, the
   :guilabel:`Version` number has been updated to `2`, matching the `V2` that appears in the
   :guilabel:`Revision` smart button of the |ECO|.

   .. image:: engineering_change_orders/BOM-version.png
      :align: center
      :alt: View current *BOM* version in the Miscellaneous tab.

.. _plm/eco/tablet-view:

Create ECO from tablet view
===========================

|ECOs| created in *tablet view* are intended for operators to directly suggest clearer operation
instructions. To create an |ECO|, navigate to the :guilabel:`Work Orders` tab on an ongoing
manufacturing order. Then, click the :guilabel:`tablet icon` for the desired work order to open the

> tablet view of the operation.

Author: Yes, "In Progress" is correct! Thanks for pointing out of necessity of navigational instructions here

.. image:: engineering_change_orders/tablet-icon.png
   :align: center
   :alt: Find the tablet icon for each operation, second from the far right.

Next, add an instructional step, by clicking the :guilabel:`☰ (three horizontal lines)` icon in the

> tablet view of an operation. Then, click the :guilabel:`Add a step` button.

Reviewer 2: This is another opportunity to expand on all the possible options that appear when this button is clicked.

Author: This is a good idea, but many of these options are related to manufacturing orders, and only the "Add a step" button is for PLM. This is a good idea to expand on these buttons in a manufacturing doc, though!

The possible options:
![image](https://github.com/odoo/documentation/assets/23611711/ff6eba9c-fe81-4b90-aaf0-102ee80eab15)

.. image:: engineering_change_orders/additional-options-menu.png
   :align: center
   :alt: Navigate to the "Add a Step" pop-up by clicking the three horizontal lines icon in tablet
         view.

> In the :guilabel:`Title` field, enter a short step description. Next, in the

Reviewer 2: This is where you should mention that clicking that reveals an "Add a Step" pop-up window, and so on...

Author: Good idea for a seamless transition after the image in the middle breaks it up (:

> :guilabel:`Instructions` text field, type the instructions of the step in greater detail.

Reviewer 2: I believe it's just 'Instruction' field, right?

Optionally, add an image to the :guilabel:`Document` field. Once completed, finish by clicking the
:guilabel:`Propose Change` button.

.. example::
   To propose an additional check for broken components, enter the details in the :guilabel:`Add a
   Step` pop-up window. Doing so creates an instructional quality control point that will be
   reviewed in the following section.

   .. image:: engineering_change_orders/add-a-step.png
      :align: center
      :alt: Fill out the *Add a Step* form to suggest an additional quality control point.

View ECO
--------

> To review the proposed changes, go to the `BOM Changes` ECO type from :menuselection:`PLM app -->

Reviewer 2: I'm only seeing BOM Updates on the PLM Overview page...is that the same thing?

Author: Yup! The cards on the PLM Overview page are like sales teams or projects. Everyone's ECO types can be different. From what I've seen, our runbots have the New Product Introduction and BOM updates eco types created by default, but sometimes they're deleted...

      :align: center
      :alt: Find the new ECO in the "BOM Changes" ECO type, in the *New* stage.

On the new |BOM|, switch to the :guilabel:`Operations` tab, and select the :guilabel:`☰ (Show
Instructions)` icon. Doing so opens a list of :guilabel:`Steps` to perform the operation, with the
newest instruction titled `New Step Suggestion:`, followed by the user-entered title. Click the line
item to view the suggested changes.

.. image:: engineering_change_orders/show-instructions.png
   :align: center
   :alt: "Show Instructions" icon in the *Operations* tab of a BoM.

On the :ref:`quality control point <quality/quality_control_points>` form, ensure the following form
fields are accurately filled out to give detailed instructions for operators:

- :guilabel:`Title`: rename to give a concise description of the new instruction
- :guilabel:`Control per`: using the drop-down menu, determine whether this instruction applies
  broadly for the :guilabel:`Product`, specifically for this :guilabel:`Operation` *only*, or a
  particular :guilabel:`Quantity` of the product
- :guilabel:`Type`: categorizes the control point type. From the drop-down menu, select

>   :guilabel:`Instructions` to detail an instruction for the worker. To receive input from the

Reviewer 2: I'm not seeing an Instructions field on my run-through. Also, it may be best to just do a brief breakdown of every field (for instance, I don't see any explanation for the Team or Responsible fields, etc.)

Author: This is tough, because this form is technically a quality control point form, and I'm not sure whether I should be exhaustive with the explanation as it's covered in the Quality documentation.  What do you think? Should I still rehash this in the context of PLM?

  workers, select the :guilabel:`Take a Picture`, :guilabel:`Register Consumed Materials`,
  :guilabel:`Print Label`, or other :ref:`quality check options <quality/quality_control_points>`.

Once the quality control point is configured, return to the :guilabel:`Steps` list using the
breadcrumbs. Finally, drag the last quality control line item to its intended order of instructions.

.. example::
   Drag and reorder the `Check for broken switches` instruction, by clicking and dragging its "6
   dots" icon to move it from the bottom to the second position.

   .. image:: engineering_change_orders/reorder.png
      :align: center
      :alt: Drag and reorder instructions by selecting the "6 dots" icon on the far left.

.. _plm/eco/eco-type:

Create ECO type
===============

To access and manage ECO types, navigate to :menuselection:`PLM app --> Configuration --> ECO
Types`.

> 

Reviewer 2: a screenshot of a blank ECO Type form could be beneficial here.

Reviewer 3: To expand on that, try to fill in forms with relevant example data when possible to visually demonstrate *how* to fill in the form, and with *what* types of information 👍

Create a new ECO type by clicking :guilabel:`New`, and fill in the :guilabel:`Name`. The
:guilabel:`Email Alias` field is optional, but feedback submitted to the filled-in email address
will automatically create an |ECO| in this ECO type.

> Modify existing ECO type names and email aliases by clicking on the respective ECO type in the list.

Reviewer 2: ...found where?

Existing ECO types are displayed in the task form, on the :guilabel:`PLM Overview` page, which is
accessible by navigating to :menuselection:`PLM app --> Overview`.

.. _plm/eco/stage-config:

Stage configuration
-------------------

Click an ECO type from :menuselection:`PLM app --> Overview` to open a kanban view of |ECOs| of this
type.

> In the pipeline, ensure there is at least one *verification* stage, where |ECO| changes can be

Reviewer 2: might be beneficial to explain how to ensure such a thing

applied, and a *closing stage*, for storing completed |ECOs|.

To configure a stage, hover over the intended stage, and select the :guilabel:`⚙️ (gear)` icon.
Then, click :guilabel:`Edit` to open a pop-up window.

> 

Reviewer 2: a screenshot of the Edit: New pop-up window that appears when the gear icon is clicked would be useful

Configure the verification stage in the edit stage pop-up window, by checking the box for
:guilabel:`Allow to apply changes`.

Optionally, add :guilabel:`Approvals` to automatically notify people to verify the changes before
they go live. Once finished, select the :guilabel:`Save & Close` button.

> 

Reviewer 2: I think it would be best to expand on every field/option that appears in this pop-up (and every other pop-up and form)

=========================
Engineering change orders
=========================

.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`
.. |BOMs| replace:: :abbr:`BoMs (Bills of Materials)`
.. |ECO| replace:: :abbr:`ECO (Engineering Change Order)`
.. |ECOs| replace:: :abbr:`ECOs (Engineering Change Orders)`

.. _plm/eco:

Utilize *engineering change orders* (*ECOs*) to track, implement, and revert change versions made to
products, and :ref:`bills of materials <manufacturing/management/bill-configuration>`.

Engineering change orders can be created:

> #. :ref:`directly in the ECO type <plm/eco/create-eco>`
> #. by an operator in the :ref:`tablet view <plm/eco/tablet-view>` of an operation
> #. automatically from feedback submitted to the :ref:`ECO type's email alias <plm/eco/eco-type>`.

Author: ```suggestion
#. :ref:`directly in the ECO type <plm/eco/create-eco>`.
#. by an operator in the :ref:`tablet view <plm/eco/tablet-view>` of an operation.
#. automatically from feedback submitted to the :ref:`ECO type's email alias <plm/eco/eco-type>`.
```

   orders. Doing so ensures employees only view the |ECOs| related to their responsibilities,
   whether it involves new product introductions, targeted product line updates, or regulatory
   compliance fulfillment.

On the |ECO| form, fill in the following fields accordingly:

- :guilabel:`Description` is a brief summary of the improvement.
- :guilabel:`Type`: specifies the |ECO| type project for organizing the |ECOs|.
- :guilabel:`Apply on` determines if the |ECO| changes the :guilabel:`Bill of Materials` or the
  :guilabel:`Product Only`.
- :guilabel:`Product` indicates the product being improved.
- :guilabel:`Bill of Materials` specifies the changed |BOM|. It auto-populates if the product in
  :guilabel:`Product` has an existing |BOM|. If multiple |BOMs| exist, select the intended radio
  options from the drop-down menu.
- :guilabel:`Company` field is used in multi-company databases. Specify if the change applies to
  products in a specific company, or leave blank if the change applies to all companies.
- :guilabel:`Responsible`: assignee in charge of this |ECO|. (Optional)
- :guilabel:`Effective`: specifies when the |ECO| becomes live. Choosing :guilabel:`As soon as

>   possible` means that the |ECO| applies to the production |BOM| as soon as an authorized user

Reviewer 2: ```suggestion
  possible` means the |ECO| applies to the production |BOM| as soon as an authorized user
```

>   :ref:`applies the changes <plm/eco/apply-changes>`. On the other hand, choosing :guilabel:`At
>   Date` and setting a specific date leaves a date that makes it easier to track the version history
>   of the |BOM| and tracking the specific date |BOMs| are used for production.

Reviewer 2: ```suggestion
  :ref:`applies the changes <plm/eco/apply-changes>`.
  
  On the other hand, choosing :guilabel:`At Date`, and setting a specific date, leaves a date that
  makes it easier to track the version history of the |BOM|, and the specific date |BOMs|, used for
  production.
```

- :guilabel:`Tags` are assigned to |ECOs| for prioritization and organization. Create a new tag by
  typing the name in the field and selecting :guilabel:`Create` from the drop-down menu. (Optional)

After filling out the |ECO| form, click the :guilabel:`Start Revision` button to begin implementing
the changes.

By pressing :guilabel:`Start Revision`, three actions occur:

#. The :guilabel:`Documents` smart button appears, storing relevant files of the |BOM|.

> #. A copy of the production |BOM| is stored in the newly appeared :guilabel:`Revision` smart button

Reviewer 2: ```suggestion
#. A copy of the production |BOM| is stored in the newly-appeared :guilabel:`Revision` smart button
```

   of the |ECO|. The next available version number (e.g., V2, V3...) is also assigned to keep track
   of all |BOM| versions.
#. The stages of the |ECO| :guilabel:`Type` are displayed in the top-right corner of the |ECO|.

.. note::

>    The :guilabel:`Revision` smart button is available only when the :guilabel:`Bill of Materials`
>    radio button is selected in the :guilabel:`Apply on` field and the :guilabel:`Start Revision`
>    button has been pressed.

Reviewer 2: ```suggestion
   The :guilabel:`Revision` smart button is available **only** when the :guilabel:`Bill of
   Materials` radio button is selected in the :guilabel:`Apply on` field, and the :guilabel:`Start
   Revision` button has been pressed.
```

.. image:: engineering_change_orders/eco-form.png
   :align: center
   :alt: ECO with overview of stages in the top-right corner, and *Revision* smart button.

Change components
-----------------

To modify the components in a |BOM|, click the :guilabel:`Revision` smart button on an |ECO| to
access the new version of the |BOM|. Odoo distinguishes the non-production version of the |BOM| from
the current version, by flagging the test |BOM| with a large :guilabel:`Archived` tag.

.. example::
   After clicking the :guilabel:`Start Revision` button for an |ECO| for the product, `[D_0045
   Stool]`, make changes to the product's |BOM| by clicking the :guilabel:`Revision` smart button.
   Doing so opens the archived |BOM|, marked with a large red :guilabel:`Archived` flag.

   .. image:: engineering_change_orders/archived-bom.png
      :align: center
      :alt: Show the archived Bill of Materials.

> On the new |BOM|, in the :guilabel:`Components` tab. Modify the components list by changing the
> :guilabel:`Quantity` of existing components, add new components using the :guilabel:`Add a line`
> button, and remove components with the :guilabel:`🗑️ (trash)` icon.

Reviewer 2: ```suggestion
On the new |BOM|, in the :guilabel:`Components` tab, proceed to modify the components list, by
changing the :guilabel:`Quantity` of existing components, adding new components using the
:guilabel:`Add a line` button, and removing components with the :guilabel:`🗑️ (trash)` icon.
```

.. _plm/eco/example-keyboard:

.. example::
   In version two of the |BOM| for a keyboard, the component quantities are reduced, and an
   additional component, `Stabilizers`, is added.

   .. image:: engineering_change_orders/version-2-bom.png
      :align: center
      :alt: Make changes to components by going to the new BoM with the *Revision* smart button.

Once the changes are complete, navigate back to the |ECO|, by clicking `ECO00X` in the breadcrumbs
located in the top-left corner. On the |ECO| form, a new :guilabel:`BoM Changes` tab displays the
differences between the current |BOM| and the new version.

Blue text indicates components newly added to the revised |BOM| that do not yet exist in the
production |BOM|. Black text represents updates shared by both |BOMs|, while red text indicates

> components present in the production |BOM| but removed in the revised |BOM|.

Reviewer 2: ```suggestion
components present in the production |BOM|, but removed in the revised |BOM|.
```

for the |BOM|. To make changes, select each operation line item, which opens the corresponding
:guilabel:`Open: Operations` pop-up window.

.. note::
   The :guilabel:`Operations` tab is *not* available by default. To enable it, navigate to
   :menuselection:`Manufacturing app --> Configuration --> Settings`, and check the :guilabel:`Work
   Orders` box.

Make changes to any of the fields in the :guilabel:`Open: Operations` pop-up window, then click
:guilabel:`Save` once completed.

Create new operations by clicking the :guilabel:`Add a line` button, and remove new operations by
clicking the :guilabel:`Archive Operation` button.

Once the changes are complete, navigate back to the |ECO|, by clicking `ECO00X` in the breadcrumbs
located in the top-left corner.

On the |ECO| form, a new :guilabel:`Operation Changes` tab displays the differences between the
current production |BOM| and the new version.

Blue text indicates operations newly added to the revised |BOM| that do not yet exist in the
production |BOM|. Black text represents updates shared by both |BOMs|, while red text indicates

> operations present in the production |BOM| but removed in the revised |BOM|.

Reviewer 2: ```suggestion
operations present in the production |BOM|, but removed in the revised |BOM|.
```

Modifications to the |BOM| in an |ECO| will **not** affect the |BOM| used in production. That is,
until the :ref:`changes are applied <plm/eco/apply-changes>`.

> In the :guilabel:`Operation Changes` tab, each row of details the columns in the table reflect the
> following information:

Reviewer 2: ```suggestion
In the :guilabel:`Operation Changes` tab, each row of details, beneath the columns in the table,
reflect the following information:
```

- :guilabel:`Operation`: Name of the operation that was modified.

> - :guilabel:`Step` column specifies the quality control point, visible when the operation includes
>   detailed instructions.

Reviewer 2: ```suggestion
- :guilabel:`Step`: specifies the quality control point, visible when the operation includes
  detailed instructions.
```

.. note::
   To check for instructions, click the operation line item in the :guilabel:`Operations` tab of a

>    |BOM|. Then, in the :guilabel:`Open: Operations` pop-up, look for the :guilabel:`Instructions`

Reviewer 2: ```suggestion
   |BOM|. Then, in the :guilabel:`Open: Operations` pop-up window, look for the :guilabel:`Instructions`
```

   smart button displayed at the top.

.. example::
   The `Assembly` :guilabel:`Operation` includes `10` detailed :guilabel:`Instructions` to
   complete it.

   .. image:: engineering_change_orders/instructions-smart-button.png
      :align: center
      :alt: Show *Instructions* smart button to check whether an operation has additional
            instructions.

> - :guilabel:`Step Type` details the type of quality control for further instructions in the
>   operation.
> 
> - :guilabel:`Type` field corresponds with the colored text to specify how the revised |BOM| differs
>   from the production |BOM|. Operation change types can be :guilabel:`Add`, :guilabel:`Remove`, or
>   :guilabel:`Update`.
> - :guilabel:`Work Center`: specifies the work center the operation is performed at.
> - :guilabel:`Manual Duration Change`: refers to the change in the :guilabel:`Default Duration` field
>   in the :guilabel:`Open: Operations` pop-up, which specifies the expected time for completing the
>   operation.

Reviewer 2: ```suggestion
- :guilabel:`Step Type` details the type of quality control for further instructions in the
  operation.
- :guilabel:`Type` corresponds with the colored text to specify how the revised |BOM| differs
  from the production |BOM|. Operation change types can be :guilabel:`Add`, :guilabel:`Remove`, or
  :guilabel:`Update`.
- :guilabel:`Work Center` specifies the work center at which the operation is performed.
- :guilabel:`Manual Duration Change` refers to the change in the :guilabel:`Default Duration` field
  in the :guilabel:`Open: Operations` pop-up window, which specifies the expected time for completing the
  operation.
```

.. example::
   The :guilabel:`Operation Changes` tab compares the production |BOM| with the revised |BOM| in the
   |ECO|.

   In the revised |BOM|, a new `Assembly` :guilabel:`Operation` at the :guilabel:`Work Center`
   `Assembly Line 1` is added. In addition, the expected duration of the operation is `20.00`
   minutes, as specified by the :guilabel:`Manual Duration Change`.

   To supplement the `Assembly` operation, two quality control point instructions are added:

   #. The first is the :guilabel:`Step` `QCP00039`, a :guilabel:`Step Type` to :guilabel:`Register
      Production` of components.
   #. The second :guilabel:`Step` is `QCP00034`, an `Instructions` :guilabel:`Step Type` that
      provides additional assembly details.

   .. image:: engineering_change_orders/operation-changes.png
      :align: center
      :alt: Show *Operation Changes* tab in an |ECO|.

.. _plm/eco/apply-changes:

Apply changes
=============

After verifying the changes, move the |ECO| to a :ref:`verification stage <plm/eco/stage-config>`,

> which are stages that require approval before the revised changes can be applied to production
> |BOM|.

Reviewer 2: ```suggestion
which are stages that require approval before the revised changes can be applied to the
production |BOM|.
```

Once the approvers accept the changes, the :guilabel:`Apply Changes` button becomes available. Click
this button, and the |ECO| is automatically moved to a closing stage. The changes are applied, which

> archives the original, production |BOM|, and the revised |BOM| becomes the new production |BOM|.

Reviewer 2: ```suggestion
archives the original production |BOM|, and the revised |BOM| becomes the new production |BOM|.
```

Verify changes
--------------

> To ensure the changes are live, from the |ECO| that where the :guilabel:`Apply Changes` button was
> just pressed, return to the revised |BOM| by clicking the :guilabel:`Revision` smart button.

Reviewer 2: ```suggestion
To ensure the changes are live, from the |ECO| where the :guilabel:`Apply Changes` button was
just pressed, return to the revised |BOM| by clicking the :guilabel:`Revision` smart button.
```

   :menuselection:`PLM app --> Overview`. By default, |ECOs| created from tablet view are set to
   spawn in the `New` stage.

   The :guilabel:`Responsible` field is assigned to the operator who made the suggestion, allowing
   the employee revising the |BOM| to seek further clarification from the person who proposed the
   changes.

   .. image:: engineering_change_orders/view-bom-change.png
      :align: center
      :alt: Find the new ECO in the "BOM Changes" ECO type, in the *New* stage.

On the revised |BOM|, switch to the :guilabel:`Operations` tab, and select the :guilabel:`☰ (Show
Instructions)` icon. Doing so opens a list of :guilabel:`Steps` to perform the operation, with the
newest instruction titled `New Step Suggestion:`, followed by the user-entered title. Click the line
item to view the suggested changes.

.. image:: engineering_change_orders/show-instructions.png
   :align: center
   :alt: "Show Instructions" icon in the *Operations* tab of a BoM.

On the :ref:`quality control point <quality/quality_control_points>` form, ensure the following form
fields are accurately filled out to give detailed instructions for operators:

> - :guilabel:`Title`: rename to give a concise description of the new instruction

Reviewer 2: ```suggestion
- :guilabel:`Title`: rename to give a concise description of the new instruction.
```
